### DISCUSSION TITLE
Play sounds from PEDs

### DISCUSSION THREAD

**Discussion by Rabtos** [YimMenu/YimMenu#2205] [Q&A] [answered]
Is it possible to play sounds from other peds?
Found this [list](https://www.jitnaught.net/gtav/natives#PLAY_PED_AMBIENT_SPEECH_NATIVE)
Tried to make a .LUA for it but I guess it won't work. I would need to implement it in the Main code, right?

```
SoundTab = gui.get_tab("GUI_TAB_FUN_VEHICLE")

script.register_looped("SetSound", function()
    if playingSound:is_enabled() then
		void AUDIO::PLAY_PED_AMBIENT_SPEECH_NATIVE(Ped TAXILIZ, const char* GENERIC_CURSE_HIGH, const char* SPEECH_PARAMS_MEGAPHONE, Any p3) // 0x8E04FEDD28D42462 0x5C57B85D b323
    end
end)

playingSound = SoundTab:add_checkbox("Enable Play Sound")
```

Can someone help me with this? I can't code and don't even know where to start to make things like this possible.

**Comment by Rabtos** [comment]
I asked ChatGPT and it gave me this:

```
Ped ped = TAXILIZ; // Set the desired Ped object
const char* speechName = "GENERIC_CURSE_HIGH"; // Set the desired ambient speech text
const char* speechParam = "SPEECH_PARAMS_MEGAPHONE"; // Set the desired speech parameters

// Function to play the sound
void playSound() {
    AUDIO::PLAY_PED_AMBIENT_SPEECH_NATIVE(ped, speechName, speechParam, 0);
}

// Somewhere in your code, when you want to play the sound
playSound();
```

If I implement this in the Main Code and rebuild YIM, would other players also be able to hear it? And I don't want the whole session to hear it, just localy.

**Reply by Yimura** [reply]
ChatGPT is not proficient in coding such specific features, especially not in a custom YimMenu Lua implementation.

The OP of the discussion has written Lua code and what you’re giving him is C/C++ code, so this is not a good answer. Please don’t rely on ChatGPT too much, it can point you in the right direction but it can’t give you a full solution.

Your C++ code in this case wouldn’t even work if you were to implement it in YimMenu directly. The way you grab the ped isn’t correct either.

I’m not trying to talk down on you or have you stop using ChatGPT but only use it once you’re confident at coding.

edit:
Just noticed you answered to your own discussion whoops.

**Comment by Yimura** [comment]
Anyways, I have a few questions that could help me to make an actual working example piece of code.

Which peds do you want to play the sound from? 1 specific ped or just all the peds around you that are loaded in?

**Reply by Rabtos** [reply]
1 specific ped, I'm not sure which one though. I visited a prostitute and just thought it would be funny if I could play those sex sounds too when doing such animations. Then had the idea with playing sounds of annoyed PEDs who yell "fuck you" or something. But didn't have any specific in mind. In the code I posted I took a random speech.
If this would work, maybe make an "fetch list" function like you can do with the animations.
Edit: @lonelybud already tested it and it doesn't work sadly.. :(

ps: Put my answer directly under your reply for better visibilty

**Reply by lonelybud** [reply]
i wanted to play police loudspeaker sound from the car. It is very cool XD.

**Reply by Rabtos** [reply]
Is it "SPEECH_PARAMS_MEGAPHONE"; ??

**Reply by lonelybud** [reply]
i dont know the purpose of speech param exactly but i know that its not the speech and just another parameter to the function call.

https://docs.fivem.net/natives/?_0x8E04FEDD28D42462

for the cop ped you need to play speech something like CHASE_VEHICLE_MEGAPHONE, CLEAR_AREA_MEGAPHONE etc

**Comment by lonelybud** [comment]
I have done this before. Players wont hear any speech (sound).

There is another code that can play sound (sound) but those sounds are limited - https://github.com/DurtyFree/gta-v-data-dumps/blob/master/soundNames.json

**Reply by lonelybud** [reply]
i tried playing sound from peds, myself from all these methods. None were heard by other players except me.
```
PLAY_PED_AMBIENT_SPEECH_NATIVE
PLAY_PED_AMBIENT_SPEECH_AND_CLONE_NATIVE
PLAY_PED_AMBIENT_SPEECH_WITH_VOICE_NATIVE
```
i think they are not networked. Only this one worked -

`PLAY_SOUND_FROM_ENTITY;`

but again it does not plays the speech as asked in original discussion.

**Reply by Rabtos** [reply]
> i think they are not networked. Only this one worked

This makes me question, when playing GTA 5 Online and some PEDs are yelling at you, only you can hear them yelling? So it isn't synced at all?

**Reply by lonelybud** [reply]
this is a good question. When a ped normally in game yells at you, I believe it should be heard by all players present there. 

But if we manually do it by triggering code it does not seem to happen. 

May be someone else can answer better.

**Reply by Rabtos** [reply]
So there HAS to be some other way to trigger those sounds?

**Reply by FIying-Scotsman** [reply]
From what I remember there’s a network (not script - 2 different things) event which is sent when you “spook” peds which makes them scream and run away. 

That should be synced with other players so they see the peds running away if you scare them on your end for example. How you would manually force that event to be sent I have no idea. Probably more work that it’s worth in my opinion as the scream used won’t be consistent - just the command to play _a_ scream voice line.

**Comment by Rabtos** [comment]
> There is another code that can play sound (sound) but those sounds are limited - https://github.com/DurtyFree/gta-v-data-dumps/blob/master/soundNames.json

Yeah, I had an YimMenu version which could play some sounds. But I couldn't find it anymore. Also old versions are detected and don't support the newest version of GTA.

> I have done this before. Players wont hear any speech (sound).

Damn... :(

**Reply by lonelybud** [reply]
The speeches are in this file 

https://raw.githubusercontent.com/DurtyFree/gta-v-data-dumps/master/speeches.json

and they are played as follows for example -

```
AUDIO::PLAY_PED_AMBIENT_SPEECH_WITH_VOICE_NATIVE(m_handle, "AIMED_AT_BY_PLAYER", "ABIGAIL", "SPEECH_PARAMS_FORCE", false);
AUDIO::PLAY_PED_AMBIENT_SPEECH_NATIVE(m_handle, "PROVOKE_GENERIC", "SPEECH_PARAMS_FORCE", false);
```

**Comment by maybegreat48** [comment]
The last parameter must be set to true

**Reply by lonelybud** [reply]
for these methods?

```
PLAY_PED_AMBIENT_SPEECH_NATIVE
PLAY_PED_AMBIENT_SPEECH_AND_CLONE_NATIVE
PLAY_PED_AMBIENT_SPEECH_WITH_VOICE_NATIVE
```

are you sure? tested with other players?

**Reply by Rabtos** [reply]
Can you try this please (if you still have the code) ?

**Comment by lonelybud** [comment, marked as answer]
I have written quick code to test 

add this file in yim root- 
https://raw.githubusercontent.com/DurtyFree/gta-v-data-dumps/master/speeches.json

<details>
 <summary>
  UI
 </summary>

 ![image](https://github.com/YimMenu/YimMenu/assets/139460769/5680bde2-9ca6-4575-a9b4-8a1a37678fba)
</details>

check for the comment github-actions bot comment (Download the artifacts for this pull request) when its done
https://github.com/lonelybud/YimMenu/pull/7

**Reply by lonelybud** [reply]
others can also test.

**Reply by lonelybud** [reply]
@maybegreat48  i have passed true as last param
if still it does not work maybe then need to play with correct speechParam.
or god knows

**Reply by lonelybud** [reply]
i have actually tried this code. Most of the time it was heard by my friend. Also one need to play with speechparams.

**Reply by Rabtos** [reply]
Hey the GUI looks very nice. 
But I can't play any sounds. It says "Please select a valid ped using context menu"

![grafik](https://github.com/YimMenu/YimMenu/assets/124796345/95a68fd0-6205-49b4-aa3f-49b156148a43)

**Reply by lonelybud** [reply]
you need to check - `self > Context Menu Enabled`
then press `X` (VEH_DUCK binded key), point toward the ped (till it gets green rectangle) and then close the context menu by pressing X again.
This will select that ped.

**Reply by Rabtos** [reply]
Oooh Thank You! So I've tried it with a friend and he could hear the the first 2 Play Speech options.

```
Online + Clientside:	"PLAY_PED_AMBIENT_SPEECH_NATIVE(ped, speechName, speechParam, p3)"
Online + Clientside:	"PLAY_PED_AMBIENT_SPEECH_AND_CLONE_NATIVE(ped, speechName, speechParam, p3)"
Only Clientside:	"PLAY_PED_AMBIENT_SPEECH_WITH_VOICE_NATIVE(ped, speechName, speechParam, p4)"
Only Clientside:	"PLAY_AMBIENT_SPEECH_FROM_POSITION_NATIVE(speechName, voiceName, x, y, z, speechParam)"
```

It works, but it doesn't work like I intended. The only PED I tried was a Barber. 
Is it possible to make this work without selecting the PED ? 
And I would like to play it from my Player PED, so is there any way to make "PLAY_AMBIENT_SPEECH_FROM_POSITION_NATIVE" work Online + Clientside?

An idea would be to spawn an invisible PED right on you but I hope there are better solutions? Thank you again @lonelybud !!

**Reply by lonelybud** [reply]
let me check.

**Reply by lonelybud** [reply]
the first two i dont know seems to work same. Third one use the original voice of ped. 4th plays sound from a coord. 

So using the first one is fine when given a ped.
The 4th can be played without ped but since it is not heard online so it looks worthless.

I tried using my own character as source of ped and as predicted it does not work. So you definitely need a ped to play sound in online session.

**Reply by lonelybud** [reply]
you can also change your answer of discussion now.

**Reply by Rabtos** [reply]
Yeah just did that.
I think should open an new discussion on how to make this work better / easier.
So that you don't have to select a ped each time, it's really unpractical.
Also making the coordinates speech work online, so dat others can hear it too.

**Reply by maybegreat48** [reply]
> I tried using my own character as source of ped and as predicted it does not work

It'll work if you change your player model to MP Franklin or MP Lamar

**Reply by lonelybud** [reply]
damn XD.
